File: chor-extraction/src/main/java/ast/cc/nodes/ProcedureDefinition.java
package ast.cc.nodes;

import ast.cc.interfaces.CCNode;
import ast.cc.CCVisitor;
import epp.MergingException;

import java.util.HashSet;

public class ProcedureDefinition implements CCNode {
    private final String procedure;
    private final CCNode choreography;

    private final HashSet<String> processes;

    public ProcedureDefinition(String procedure, CCNode choreography, HashSet<String> processes) {
        this.procedure = procedure;
        this.choreography = choreography;
        this.processes = processes;
    }

    public String getProcedure() {
        return procedure;
    }

    public CCNode getChoreography() {
        return choreography;
    }

    public HashSet<String> getProcesses() {
        return processes;
    }

    @Override
    public <T> T accept(CCVisitor<T> visitor) throws MergingException {
        return visitor.visit(this);
    }
}

File: chor-extraction/src/main/java/ast/cc/nodes/ProcedureDefinition.kt
package ast.cc.nodes

import ast.cc.CCVisitor
import ast.cc.interfaces.CCNode
import epp.MergingException
import java.util.*

data class ProcedureDefinition(val procedure: String, val choreography: CCNode, val processes: HashSet<String>) : CCNode {

    @Throws(MergingException::class)
    override fun <T> accept(visitor: CCVisitor<T>): T {
        return visitor.visit(this)
    }
}
